### PR TITLE
Add pull request template 943

### LINKED ISSUE #943: Add pull request template
as per
https://github.com/fnielsen/scholia/community .

Good example in https://github.com/ytdl-org/youtube-dl/tree/master/.github .

### PR DESCRIPTION
**Pull request by DanielFLopez**
fix #943

### CODE REVIEW

**Comment by carlinmack** [comment]
Hi, taken a shot at redrafting this to be a little more relaxed and relevant to the PRs that have been made here recently. Happy to have a back and forth to develop and improve this :)

Fixes # (issue number, if applicable) 

## Description

> Please include a summary of the change, relevant motivation and context. If possible and applicable, include before and after screenshots and a URL where the changes can be seen.



---

> Please check any of the following which apply:

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
    - [ ] I have made corresponding changes to the documentation
- [ ] This change requires new dependencies (please list)

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

* Test A
* Test B

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+### Description
+> Please include a summary of the change, relevant motivation and context. If possible and applicable, include before and after screenshots and a URL where the changes can be seen.
+    
+Check any of the following which apply:
+
+* [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
+* [ ]  This change requires a documentation update
+    * [ ]  I have made corresponding changes to the documentation
```
Comment (egonw) [inline PULL_REQUEST_TEMPLATE.md:12]: I would propose to add a checkbox to encourage checking with `tox` as is (now) expected.

Comment (egonw) [reply PULL_REQUEST_TEMPLATE.md:12]: (or under the checklist below, but then I would suggest to clarify *what* new warnings the author can expect: unit tests, docs, both, etc)

**Review by egonw — CHANGES_REQUESTED** [review]
Just a small suggestion. It's good to have these templates.

@fnielsen, are there other things you like people to check before the submit a PR?

File: PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (carlinmack) [reply PULL_REQUEST_TEMPLATE.md:12]: People only have to check with tox if they change Python right? Would be good to make this clear

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+PULL REQUEST TEMPLATE
+======================
```
Comment (carlinmack) [inline PULL_REQUEST_TEMPLATE.md:2]: just to check but these lines are dropped when the template is used right?

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
```
Comment (carlinmack) [inline PULL_REQUEST_TEMPLATE.md:3]: Could you add the line:

Fixes # (issue number, if applicable)

File: PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+### Testing
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+* Test A
+* Test B
+
+### Checklist
+* [ ]  I have commented my code, particularly in hard-to-understand areas
+* [ ]  My changes generate no new warnings
+
```
Comment (carlinmack) [inline PULL_REQUEST_TEMPLATE.md:24]: ` * [ ] I have not used code from external sources without attribution`

File: PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+PULL REQUEST TEMPLATE
+======================
+
+
+### Description
+> Please include a summary of the change, relevant motivation and context. If possible and applicable, include before and after screenshots and a URL where the changes can be seen.
+
```
Comment (carlinmack) [inline PULL_REQUEST_TEMPLATE.md:7]:
```
### Caveats

> Please list anything which has been left out of this PR or which should be considered before this PR is accepted
```

**Review by carlinmack — CHANGES_REQUESTED** [review]

File: PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (carlinmack) [reply PULL_REQUEST_TEMPLATE.md:24]: `* [ ] I have considered accessibility in my implementation

Comment (carlinmack) [reply PULL_REQUEST_TEMPLATE.md:24]: `* [ ] There are no remaining debug statements (print, console.log, ...)`

File: PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (DanielFLopez) [reply PULL_REQUEST_TEMPLATE.md]: removed

File: PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (DanielFLopez) [reply PULL_REQUEST_TEMPLATE.md]: done

File: PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (DanielFLopez) [reply PULL_REQUEST_TEMPLATE.md]: done

File: PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (DanielFLopez) [reply PULL_REQUEST_TEMPLATE.md:13]: done

**Comment by DanielFLopez** [comment]
@egonw @carlinmack I have made the suggested changes, you can review them

**Review by carlinmack — APPROVED** [review]

File: PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+> Please list anything which has been left out of this PR or which should be considered before this PR is accepted
+Check any of the following which apply:
+
+* [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
+* [ ]  This change requires a documentation update
+    * [ ]  I have made corresponding changes to the documentation
+* [ ]  This change requires new dependencies (please list)
+
+if you make changes to the python code
```
Comment (Daniel-Mietchen) [inline PULL_REQUEST_TEMPLATE.md:16]: Capitalize "Python".

File: PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  
+* [ ]  my code passes [tox](https://tox.readthedocs.io/en/latest/) check, you can receive warnings about tests, documentation or both
```
Comment (Daniel-Mietchen) [inline PULL_REQUEST_TEMPLATE.md:18]: Some confusion about the intended meaning of "you" here.

File: PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+### Testing
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
```
Comment (Daniel-Mietchen) [inline PULL_REQUEST_TEMPLATE.md:21]: add period at end.

File: PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (carlinmack) [reply PULL_REQUEST_TEMPLATE.md:16]: and "IfI

Comment (carlinmack) [reply PULL_REQUEST_TEMPLATE.md:16]: oh perhaps not as I guess it is running on from the previous sentence. Maybe have it in a quote then?

`> if you make changes to the Python code:`

File: PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (DanielFLopez) [reply PULL_REQUEST_TEMPLATE.md]: I have changed it to "I"

File: PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+Check any of the following which apply:
+
+* [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
+* [ ]  This change requires a documentation update
+    * [ ]  I have made corresponding changes to the documentation
+* [ ]  This change requires new dependencies (please list)
+
+*If you make changes to the Python code*
+  
+* [ ]  my code passes [tox](https://tox.readthedocs.io/en/latest/) check, I can receive warnings about tests, documentation or both
```
Comment (Daniel-Mietchen) [inline PULL_REQUEST_TEMPLATE.md:18]: there is a `the` missing before `tox`

**Review by Daniel-Mietchen — COMMENTED** [review]
Just a minor grammatical edit needed.

**Pull request merged**